Assets/Vimeo/Scripts/Services/VimeoUploader.cs

```diff
@@ -0,0 +1,176 @@
+        public event UploadAction OnUploadError;
+        public event UploadAction OnUploadComplete;
+
+        public delegate void UploadEvent(VideoChunk chunk, string msg = "");
+        public event UploadEvent OnChunckUploadComplete;
+        public event UploadEvent OnChunckUploadError;
+
+        //Private members
+        private Queue<VideoChunk> myChunks;
+        public VimeoApi vimeoApi;
+        //TODO: In the future this will be stored in a hash table to provide batch uploading
+        private string file;
+        private string vimeo_url;
+        private FileInfo fileInfo;
+
+
+        //Public members
+        public int maxChunkSize;
+        public int numChunks;
+        private void Start()
+        {
+            this.hideFlags = HideFlags.HideInInspector;
+        }
+
+        public void Init(string _token, int _maxChunkSize = 1000)
+        {
+            //A queue of video chunks to upload
+            myChunks = new Queue<VideoChunk>();
+
+            //Instantiate the Vimeo Api
+            if (vimeoApi == null) {
+                vimeoApi = gameObject.AddComponent<VimeoApi>();
+                vimeoApi.OnError += ApiError;
+                vimeoApi.OnNetworkError += NetworkError;
+                vimeoApi.OnRequestComplete += RequestComplete;
+
+                vimeoApi.token = _token;
+            }
+
+            maxChunkSize = _maxChunkSize;
+
+        }
+        private void RequestComplete(string response)
+        {
+            string tusUploadLink = GetTusUploadLink(response);
+            vimeo_url = GetVideoPermlink(response);
+            CreateChunks(file, fileInfo, tusUploadLink);
+
+            //Kick off first chunks, others will be called on OnCompleteChunk()
+            VideoChunk firstChunk = myChunks.Dequeue();
+            firstChunk.Upload();
+        }
+        public void SetChunkSize(int size)
+        {
+            maxChunkSize = size;
+        }
+        private void ApiError(string response)
+        {
+
+        }
+        private void NetworkError(string response)
+        {
+
+        }
+        public void Upload(string _file)
+        {
+            file = _file;
+            fileInfo = new FileInfo(_file);
+
+            //Send the request, response will be catched in the RequestComplete() method
+            StartCoroutine(
+                vimeoApi.RequestTusResource("me/videos", fileInfo.Length)
+            );
+        }
+
+        private void OnCompleteChunk(VideoChunk chunk, string msg)
+        {
+            //Emit the event
+            if (OnChunckUploadComplete != null) {
+                OnChunckUploadComplete(chunk, msg);
+            }
+
+            //Destroy the chunk
+            Destroy(chunk);
+
+            //Make sure the queue is not empty
+            if (myChunks.Count != 0) {
+                VideoChunk nextChunk = myChunks.Dequeue();
+                float progres = ((float)myChunks.Count / (float)numChunks) * -1.0f + 1.0f;
```

Comment:
`progres` is spelled `progress`

also, what is `-1.0f` + `1.0f` doing

Reply:
It inverses the range so it's not 1.0f -> 0.0f. - https://github.com/vimeo/vimeo-unity-sdk/pull/46/commits/2427d90deb58cbaf846728cff0a46aec75b7e04b